src/App.vue

```diff
@@ -428,6 +428,11 @@ export default class App extends Mixins(AuthApiMixin, CommonMixin, DateMixin, Fi
       this.goToDashboard()
     })
 
+    // listen to confirm events
+    this.$root.$on('confirm-dialog', (title, message, action) => {
+      this.showConfirmDialog(title, message, action)
+    })
```

Comment:
I like the idea here but the calling code can't wait for the result of the confirm dialog. Is this more of an "acknowledgement dialog" then?

I suggest renaming this or moving the dialog to the component that uses it.

Reply:
You are right, in this scenario it is more of an acknowledgement. 
However, i did want to get away from having multiple instances of this confirm dialog and i think it still fits this purpose. 

Long term, i want to remove all instances and refs of the confirm dialog in this app, and just utilize the singular instance on the root level.

Reply:
Yes, I like the idea, but I don't think calling code can ever wait for this dialog and get a result. Events are async.

Reply:
You are probably right. 
Will look into this, if there isn't a relatively elegant solution, will remove and relocate locally into the component :)

Reply:
I tried a state approach but didn't like it. Reverted changes. 
Looking at the Mixin approach now 👀